src/RandomNumberGenerator.md

```diff
@@ -1,17 +1,118 @@
 # Random Number Generator
+
 Generates a random number within a given range
 
 ## Result
+
 ```js
-  randomNumber(1, 100); // 68
+randomNumber(1, 100); // 68
 ```
 
 ## Snip Langauge
-* [JavaScript](#javascript)
+
+- [JavaScript](#javascript)
+- [Cpp](#cpp)
+- [Python](#python)
+- [Java](#java)
+- [C#](#c)
+- [php](#php)
+- [perl](#perl)
 
 ### JavaScript
+
 ```js
-  function randomNumber(min, max) { 
-      return Math.floor(Math.random() * (max - min-1) + min+1);
-  } 
+function randomNumber(min, max) {
+  return Math.floor(Math.random() * (max - min - 1) + min + 1);
+}
+```
+
+### Cpp
+
+```cpp
+int randomNumber(int min, int max) {
+    int dummy_variable = 0;
+    unsigned long long seed = reinterpret_cast<unsigned long long>(&dummy_variable);
+
+    struct timespec time;
+    if (clock_gettime(CLOCK_REALTIME, &time) == 0) {
+        seed ^= static_cast<unsigned long long>(time.tv_sec) << 32 | time.tv_nsec;
+    }
+
+    const unsigned long long m = (1ULL << 62) - 1;
+    const unsigned long long c = 1ULL << 30;
+    const unsigned long long a = 6364136223846793005ULL;
+
+    seed = (a * seed + c) & m;
+
+    int random_number = min + (seed >> 31) % (max - min + 1);
+    return random_number;
+}
+
+```
+
+### Python
+
+```py
+def randomNumber(min_value, max_value):
```

Comment:
Still return the same number mate
![image](https://github.com/rahulrk-dev/snips/assets/47377566/a5769127-d806-47af-a012-1904d9c61a9d)